client/utils/usePersistableState.ts

```diff
@@ -0,0 +1,54 @@
+import { useState } from 'react';
+import { useBeforeUnload } from 'react-use';
+
+export const usePersistableState = <T>(
```

Comment:
This is a pattern we use everywhere, so it'll be good to have a hook for it.

Reply:
This is really cool!